### PR TITLE
🔥 Add function to override form decoder setting

### PR DESCRIPTION
**Pull request by rockcreation7**
**Please provide enough information so that others can review your pull request:**

- SetBodyParserDecoder function map all option to form decoder, which allow you to change decoder option globally, use with type BodyParserConfig, BodyParserType.

- Added Test_Ctx_BodyParser_WithSetBodyParserDecoder for testing

**Explain the *details* for making this change. What existing problem does the pull request solve?**

- You can do custom type in JSON format, Create custom type and related "UnmarshalJSON" method, But in form you need to register converter in decoder, I added option for supporting this feature. In order to solve problem when you need custom type for form.

- Allow set the decoder's zeroEmpty to true, allow not skipping empty input as it will overwrite existing value.  #1084

```
customType := BodyParserType{
		Customtype: CustomType{},
		Converter:  timeConverter,
}

SetBodyParserDecoder(
         BodyParserConfig{
		IgnoreUnknownKeys: true,
		BodyParserType:    []BodyParserType{customType},
		ZeroEmpty:         true,
		SetAliasTag:       "form",
	}
)
```

**Commit formatting** 

Use emojis on commit messages so it provides an easy way of identifying the purpose or intention of a commit. Check out the emoji cheatsheet here: https://gitmoji.carloscuesta.me/

### CODE REVIEW

**Comment by rockcreation7** [comment]
Any suggestion for moving forward on this PR will be appreciated, or any idea should be further investigate, wish to provide help, Thanks.

File: ctx.go [hunk 1]
```diff
@@ -244,6 +259,22 @@ var decoderPool = &sync.Pool{New: func() interface{} {
 	return decoder
 }}
 
+// SetBodyParserDecoder allow globally change the option of form decoder, update decoderPool
+func SetBodyParserDecoder(bodyParserConfig BodyParserConfig) {
```
Comment (ReneWerner87) [inline ctx.go:263]: can we use the method for the initial write of the decoder pool ?
https://github.com/gofiber/fiber/blob/master/ctx.go#L274
@rockcreation7

Comment (rockcreation7) [reply ctx.go:263]: Sure, I can help update the doc on this, thanks.

Please feel free to let me know anything that can help.

Comment (rockcreation7) [reply ctx.go:263]: or you mean I put a change to the initial write?

Comment (ReneWerner87) [reply ctx.go:263]: > or you mean I put a change to the initial write?
@rockcreation7 
yes exactly, with the initial set you can also use your method or ?

will merge it directly after the adjustment

**Comment by ReneWerner87** [comment]
@rockcreation7 can you rename it a bit, because the decoder is used by the query parser and body parser

maybe something just called parser config or type

**Comment by rockcreation7** [comment]
Thanks, will update.

**Comment by rockcreation7** [comment]
https://github.com/gofiber/fiber/pull/1555/

Renamed function:
SetBodyParserDecoder to SetParserDecoder

Type:
BodyParserType > ParserType
bodyParserConfig > parserConfig
BodyParserConfig > ParserConfig

**Pull request merged**